client/state/selectors/can-install-woo-on-plans.js

```diff
@@ -0,0 +1,23 @@
+import { isBusinessPlan, isEcommercePlan } from '@automattic/calypso-products';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import { getSite } from 'calypso/state/sites/selectors';
+
+/**
+ * Returns true if the current user is eligible to install WooCommerce on Plans.
+ *
+ * @param {object} state Global state tree
+ * @param {number} siteId Site ID
+ * @returns {?boolean} True if the user can participate in the free to paid upsell
+ */
+export default function canInstallWooOnPlans( state, siteId ) {
+	const productSlug = getSite( state, siteId )?.plan?.product_slug;
+
+	if ( ! productSlug ) {
+		return false;
+	}
+
+	return (
+		canCurrentUser( state, siteId, 'manage_options' ) &&
+		( isBusinessPlan( productSlug ) || isEcommercePlan( productSlug ) )
```

Comment:
Are there any selectors we can re-use here? Maybe:
- [`isSiteOnAtomicPlan`](https://github.com/Automattic/wp-calypso/blob/7655b6d1f326371990851da10ebb97f133fbdc17/client/state/selectors/is-site-on-atomic-plan.js) - probably not flexible enough to eventually include Premium, though.
- [`canUpgradeToPlan`](https://github.com/Automattic/wp-calypso/blob/7655b6d1f326371990851da10ebb97f133fbdc17/client/state/selectors/can-upgrade-to-plan.js) - more flexible but not 100% the same.

Reply:
I'd say so, probably want to check for a plan feature instead. Suspect we need to wait for dusty's diff or implement something in calypso for it too. I'm assuming we can refactor this in later.